src/mbgl/style/layer.cpp

```diff
@@ -148,5 +148,27 @@ bool LayerFactory::initSourceLayerAndFilter(Layer* layer, const conversion::Conv
     return true;
 }
 
-} // namespace style
+std::unique_ptr<style::Layer> LayerManager::createLayer(
+    const std::string& type, const std::string& id,
+    const style::conversion::Convertible& value, style::conversion::Error& error) noexcept {
+    if (LayerFactory* factory = getFactory(type)) {
+        auto layer = factory->createLayer(id, value);
+        if (!layer) {
+            error.message = "Error parsing a layer of type: " + type;
+        }
+        return layer;
+    }
+    error.message = "Unsupported layer type: " + type;
+    return nullptr;
+}
+
+std::unique_ptr<RenderLayer> LayerManager::createRenderLayer(Immutable<style::Layer::Impl> impl) noexcept {
+    if (LayerFactory* factory = getFactory(impl->getTypeInfo())) {
```

Comment:
Is this `if` needed? We could have just:
```
LayerFactory* factory = getFactory(impl->getTypeInfo());
assert(factory);
return factory->createRenderLayer(std::move(impl));
```

Reply:
Nice catch 👍 Indeed, unlike the style parsing case, here there must be a factory for an `Immutable<style::Layer::Impl>` instance.